### PR TITLE
[SPARK-28208][SQL][FOLLOWUP] Use `tryWithResource` pattern

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to use `tryWithResource` for ORC file.

### Why are the changes needed?

This is a follow-up to address https://github.com/apache/spark/pull/25006#discussion_r298788206 .

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Pass the Jenkins with the existing tests.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @HyukjinKwon

**Review by HyukjinKwon — APPROVED** [review]
LGTM if tests pass

**Comment by dongjoon-hyun** [comment]
Thank you for review. I transformed the test cases, too.

**Comment by dongjoon-hyun** [comment]
Retest this please.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala [hunk 1]
```diff
@@ -180,11 +179,12 @@ class OrcFileFormat
 
       val fs = filePath.getFileSystem(conf)
       val readerOptions = OrcFile.readerOptions(conf).filesystem(fs)
-      val reader = OrcFile.createReader(filePath, readerOptions)
-
-      val requestedColIdsOrEmptyFile = OrcUtils.requestedColumnIds(
-        isCaseSensitive, dataSchema, requiredSchema, reader, conf)
-      reader.close()
+      val requestedColIdsOrEmptyFile = {
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala]: Nit: The outer braces are redundant here

Comment (dongjoon-hyun) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala]: Thanks!

**Review by HeartSaVioR — APPROVED** [review]
LGTM

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon , @srowen , @HeartSaVioR . Merged to master.

**Pull request closed without merging**